### ISSUE TITLE
0s... FAIL matrix-lib-test when cross-compiling

### ISSUE THREAD

**Issue by lordadamson** [kaldi-asr/kaldi#4137]
So, some libraries have tests that run while building. Which is a problem if you're cross-compiling. I assume the issue is that the tests fail because it's built for a different architecture and the build system is trying to run it on the host.

unfortunately that disrupts the build process.

My workaround currently is that I'm doing this before I `make`

```bash
sed -i 's|TESTFILES|#TESTFILES|' base/Makefile
sed -i 's|TESTFILES|#TESTFILES|' matrix/Makefile
```

These two commands will comment the two lines that specify the tests that should be run in base and matrix (these are the ones that fail on my machine).

So maybe provide an option so that I can configure kaldi not to run the tests while building. That would be awesome.

**Comment by kkm000** [comment]
I'm currently looking into another cross-platform issue with configure, let me keep that in mind. You are absolutely right, we should not try to run tests compiled for Android... :-\

**Comment by lordadamson** [comment]
Yeah sure. Thanks for responding. I can work on this if you want me to.

**Comment by kkm000** [comment]
I need to do the configure part first, to make the TARGET_ARCH variable
available in the Makefile. After that it's no brainer, so do not bother.

**Comment by lordadamson** [comment]
Alright cool :D

**Comment by kkm000** [comment]
@lordadamson, this is next in my pipeline. I'll ask you to review and try the changes--hope you'll be able to find time for that?

**Comment by lordadamson** [comment]
Sure! I'll be happy to help.

**Comment by yuezhangW** [comment]
it's useful! Much thanks!!!

**Comment by greeeenmouth** [comment]
It's helpful. Thank you so much :)
